src/Internal/Payment/State/VerifiedState.php

```diff
@@ -0,0 +1,72 @@
+<?php
+/**
+ * Class VerifiedState
+ *
+ * @package WooCommerce\Payments
+ */
+
+namespace WCPay\Internal\Payment\State;
+
+use WCPay\Constants\Payment_Intent_Status;
+use WCPay\Core\Exceptions\Server\Request\Extend_Request_Exception;
+use WCPay\Core\Exceptions\Server\Request\Immutable_Parameter_Exception;
+use WCPay\Core\Exceptions\Server\Request\Invalid_Request_Parameter_Exception;
+use WCPay\Internal\Service\PaymentRequestService;
+
+/**
+ * This state is used to create payment intent and verify next actions based on the intent state.
+ */
+class VerifiedState extends AbstractPaymentState {
```

Comment:
> But it feels like a redundant step a little bit. Most state transitions will probably require some sort of data verification as they progress through the process, so I think it might make sense to roll Verified back to Initial.

The state is not redundant. See https://github.com/Automattic/woocommerce-payments/issues/7411 and specifically "Handling existing successful intents". This is because of the way the duplicate payment prevention service works now, and needs to work, at least until we implement storage for the payment context. If there was an existing intent, detected during the verification stage, payment processing needs to be skipped, and we'd transition straight to the `ProcessedState`.

Now that I'm writing this though, I realize that we can simply do this within `InitialState`:

```php
class InitialState {
	function start_processing() {
		// prepare data...
		
		// verification
		$existing_intent = $this->duplicate_payment_prevention_service->check_for_existing_intent();
		if ( $existing_intent ) {
			$context->set_intent( $existing_intent );
			$next_state = $this->create_state( ProcessedState::class );
			$next_state->complete();
		}

		// normal processing (what we have in VerifiedState now)
		$intent = $this->payment_request_service->create_intent( $context );
		$context->set_intent( $intent );

		// Intent requires authorization (3DS check).
		if ( Payment_Intent_Status::REQUIRES_ACTION === $intent->get_status() ) {
			return $this->create_state( AuthenticationRequiredState::class );
		}

		// All good. Proceed to processed state.
		$next_state = $this->create_state( ProcessedState::class );
		return $next_state->complete();
	}
}
```

@zmaglica splitting `InitialState` into `InitialState` and `VerifiedState` was not a part of the issue definition for this PR. Why did you go ahead and implement it?

Reply:
> @zmaglica splitting `InitialState` into `InitialState` and `VerifiedState` was not a part of the issue definition for this PR. Why did you go ahead and implement it?

The reason for splitting lies in the introduction of new states. For more details about this decision, please refer to the conversation at: p1697092433168499/1696947391.111769-slack-C05997J5PCG. As you follow the conversation, you will notice that these new states have been discussed/added, and they need to be defined before we can proceed with defining the `AuthenticationRequiredState`. The only state that is not part of the authentication flow, which is part of this PR, is the `ProcessedState`, but I didn't want to leave the processing flow half-done.

Reply:
> For more details about this decision, please refer to the conversation at: p1697092433168499/1696947391.111769-slack-C05997J5PCG. As you follow the conversation, you will notice that these new states have been discussed/added, and they need to be defined before we can proceed with defining the `AuthenticationRequiredState`.

I'm too used to having the verified state, and I mistakenly used as a part of the example in this message (p1697092433168499/1696947391.111769-slack-C05997J5PCG). However, it seems like we do not need the state for now, so let's merge it back with `InitialState`, please. @htdat can re-add it in https://github.com/Automattic/woocommerce-payments/pull/7450 if needed, but that will probably not be the case. Sorry for triggering this change, and then requesting it to be reverted.

> the only state that is not part of the authentication flow, which is part of this PR, is the ProcessedState, but I didn't want to leave the processing flow half-done.

There will be more states, as we proceed with the payment process. I understand the temptation to create them all in advance, but that's not needed yet. Let's leave the `ProcessedState`, but try to merge this PR soon, to allow @htdat to use it within the in-progress PR.

--- 

Regarding @marcinbot's request, lets rename:

- `AuthenticationRequiredState` to `PendingAuthenticationState`
- `CompletedState` to `CapturedState`
- Introduce a new (and empty for now) `PendingCaptureState` state. Based on the outcome of the intent (captured or not), `ProcessedState` should transition to either this state, or `CapturedState`.

@marcinbot how would we call the transition? `complete_processing()` instead of `complete()`?

Reply:
I have removed VerifiedState here: https://github.com/Automattic/woocommerce-payments/pull/7471/commits/7765bee4c4e6a445879d0019d080de8be2660c52